bedrock/wordpress/models.py

```diff
@@ -0,0 +1,113 @@
+import operator
+from django.conf import settings
+from django.db import models, transaction
+from django.db.models import Q
+from django.db.utils import DatabaseError
+from django.utils.html import strip_tags
+
+from django_extensions.db.fields.json import JSONField
+from jinja2 import Markup
+
+from bedrock.wordpress.api import get_posts_data
+
+
+class BlogPostQuerySet(models.QuerySet):
+    def filter_by_blog(self, blog_slug):
+        return self.filter(wp_blog_slug=blog_slug)
+
+    def filter_by_tags(self, *tags):
+        tag_qs = [Q(tags__contains='"{}"'.format(t)) for t in tags]
+        return self.filter(reduce(operator.or_, tag_qs, Q()))
+
+
+class BlogPostManager(models.Manager):
+    def get_queryset(self):
+        return BlogPostQuerySet(self.model, using=self._db)
+
+    def filter_by_blog(self, blog_slug):
+        return self.get_queryset().filter_by_blog(blog_slug)
+
+    def filter_by_tags(self, *tags):
+        return self.get_queryset().filter_by_tags(*tags)
+
+    def update_posts(self, data, database='default'):
+        with transaction.atomic(using=database):
+            count = 0
+            posts = data['posts']
+            self.filter_by_blog(data['wp_blog_slug']).delete()
+            for post in posts:
+                try:
+                    self.create(
+                        wp_id=post['id'],
+                        wp_blog_slug=data['wp_blog_slug'],
+                        date=post['date_gmt'],
+                        modified=post['modified_gmt'],
+                        title=post['title']['rendered'],
+                        excerpt=post['excerpt']['rendered'],
+                        link=post['link'],
+                        featured_media=post['featured_media'],
+                        tags=post['tags'],
+                    )
+                    count += 1
+                except DatabaseError:
+                    continue
+
+            return count
+
+    def refresh(self, feed_id, database='default', num_posts=None):
+        data = get_posts_data(feed_id, num_posts)
+        if data:
+            return self.update_posts(data, database)
+
+        # no data returned. something wrong.
+        return 0
+
+
+class BlogPost(models.Model):
+    wp_id = models.IntegerField()
+    wp_blog_slug = models.CharField(max_length=50)
+    date = models.DateTimeField()
+    modified = models.DateTimeField()
+    title = models.CharField(max_length=255)
+    excerpt = models.TextField()
+    link = models.URLField()
+    featured_media = JSONField()
+    tags = JSONField()
+
+    objects = BlogPostManager()
+
+    class Meta:
+        get_latest_by = 'date'
+        ordering = ['-date']
+
+    def __unicode__(self):
+        return '%s: %s' % (self.blog_name, self.title)
```

Comment:
Line https://github.com/mozilla/bedrock/pull/4757/files#diff-71ab2ee0e7addf70ea176c9c9c18fdc7R19 should also need `u`

Or better use `unicode_literals` as we do in other places to avoid running behind those `str` vs `unicode` errors.